clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -280,7 +295,27 @@ void Sema::checkSYCLDeviceVarDecl(VarDecl *Var) {
   SourceRange Loc = Var->getLocation();
   llvm::DenseSet<QualType> Visited;
 
-  checkSYCLVarType(*this, Ty, Loc, Visited);
+  checkSYCLVarType(*this, Ty, Loc, Visited, false);
+}
+
+static void checkAccessorVarType(Sema &S, QualType Ty, SourceRange Loc) {
+  assert(Util::isSyclAccessorType(Ty) &&
+         "Should only be called on SYCL accessor types.");
+
+  llvm::DenseSet<QualType> Visited;
+  const RecordDecl *RecTy = Ty->getAsRecordDecl();
+  for (const auto &Field : RecTy->fields()) {
+    if (Field->getName().compare("MData") == 0) {
```

Comment:
That's the member of the accessor data representation that has the pointer to the data.

Reply:
Its already bad enough that we had to teach the CFE the NAME of the accessor type, we shouldn't be constraining the implementation names of it.  If the library ever chooses to name the field correctly (it likely needs to be _MData to prevent macro-shennanagins), they'd have to change it here too, and I don't see a way of detecting that.

Can we figure out the field based on its type or contents rather than name?

Reply:
AFAIK first template parameter of accessor class describes type of data.

Reply:
I switched this to use the type of the first argument passed to the template. So no more tight coupling with names inside the accessor.